src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -94,17 +94,42 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
         ChipLogDetail(Crypto, "AES_CCM_encrypt: Using aad == null path");
     }
 
-    if (plaintext_length != 0)
+    // psa_aead_update() requires the usage of the macro PSA_AEAD_UPDATE_OUTPUT_SIZE to determine the output buffer size.
+    // For AES-CCM, PSA_AEAD_UPDATE_OUTPUT_SIZE will round up the size to the next multiple of the block size (16).
+    // If the ciphertext length is not a multiple of the block size, we will encrypt in two steps, first with the
+    // block_aligned_length, and then with a rounded up partial_block_length, where a temporary buffer will be used for the output.
+    constexpr uint8_t kBlockSize = 16;
```

Comment:
This 16 must rather come from a PSA macro, if one is available, such as a macro/constant from PSA_AEAD_UPDATE_OUTPUT_SIZE